proto/teleport/lib/teleterm/v1/tshd_events_service.proto

```diff
@@ -44,6 +44,22 @@ service TshdEventsService {
   // code.
   rpc PromptMFA(PromptMFARequest) returns (PromptMFAResponse);
 
+  // PromptHardwareKeyPINAsk notifies the Electron app that the daemon is waiting for the user to
+  // provide the hardware key PIN.
+  rpc PromptHardwareKeyPINAsk(PromptHardwareKeyPINAskRequest) returns (PromptHardwareKeyPINResponse);
+
+  // PromptHardwareKeyTouch notifies the Electron app that the daemon is waiting for the user to
+  // provide the hardware key touch.
```

Comment:
>I assume that the Electron app returns a response immediately because the actual handling of a hardware key touch is handled by tshd, right?

The touch is handled by tshd, but Electron app doesn't return immediately, since this would mean closing the dialog. Instead, the app just waits until the daemon cancels the RPC. I added a comment to make it more clear.